feat(provider/kubernetes): runJob more fields

ethanfrogers opened the pull request:
add support for more of the Pod spec within the Run Job stage. also adds
a couple new components that could be dropped in to replace what is
currently in the container configurer.

spinnaker/spinnaker#1961

ethanfrogers commented:
![image](https://user-images.githubusercontent.com/3324110/32792966-b0dad2b2-c932-11e7-83ae-7e836abbbd0f.png)

danielpeach commented on app/scripts/modules/kubernetes/src/container/ports.component.html:
```diff
@@ -0,0 +1,55 @@
+</div>
+<button class="add-new btn btn-block btn-sm" ng-click="$ctrl.addPort()">
+  <span class="glyphicon glyphicon-plus-sign"></span> Add New Port
+</button>
```
New line at the end of this template.

danielpeach commented on app/scripts/modules/kubernetes/src/container/ports.component.js:
```diff
@@ -0,0 +1,34 @@
+
+const angular = require('angular');
+
+module.exports = angular.module('spinnaker.deck.kubernetes.container.ports.component', [])
```
I think some modules are named `spinnaker.deck.kubernetes...`, but they should just be `spinnaker.kubernetes...`.

danielpeach commented on app/scripts/modules/kubernetes/src/container/ports.component.js:
```diff
@@ -0,0 +1,34 @@
+      ports: '='
+    },
+    templateUrl: require('./ports.component.html'),
+    controller: function(){
```
Missing space: `function() { ... }`

danielpeach commented on app/scripts/modules/kubernetes/src/container/resources.component.html:
```diff
@@ -0,0 +1,42 @@
+<div class="form-group">
+  <div class="col-md-3">
+  </div>
+  <div class="col-md-4 sm-label-left">
```
I think this should be a `col-md-7`, and you should be able to remove the `div` above. 

The other thing to try would be `<div class="col-md-4 col-md-offset-3 sm-label-left">...</div>` (and still remove the `div` above).

danielpeach commented on app/scripts/modules/kubernetes/src/pipeline/stages/runJob/configureJob.controller.js:
```diff
@@ -1,12 +1,17 @@
 'use strict';
 
 const angular = require('angular');
+import {KUBERNETES_LIFECYCLE_HOOK_CONFIGURER} from '../../../container/lifecycleHook.component';
```
Spaces around the import (`{ KUBERNETES_LIFECYCLE_HOOK_CONFIGURER }`), and you should be able to make the import path `'kubernetes/container/lifecycleHook.component'`.

danielpeach commented on app/scripts/modules/kubernetes/src/pipeline/stages/runJob/configureJob.controller.js:
```diff
@@ -133,6 +138,20 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
       }
     };
 
+    this.setPostStartHandler = function(handler) {
```
Use arrow functions here and below (especially since these functions use `this`).

ethanfrogers replied:
good catch. that was a copy/paste oversight!

danielpeach commented on app/scripts/modules/kubernetes/src/pipeline/stages/runJob/configureJob.html:
```diff
@@ -62,15 +62,63 @@
-  
+        
+        <stage-config-field label="DNS Policy">
+          <div>
```
Is this div needed?

danielpeach commented on app/scripts/modules/kubernetes/src/pipeline/stages/runJob/configureJob.html:
```diff
@@ -62,15 +62,63 @@
+        </stage-config-field>
+
+        <stage-config-field label="Service Account Name">
+          <div>
```
Same question here.

danielpeach commented on app/scripts/modules/kubernetes/src/pipeline/stages/runJob/configureJob.html:
```diff
@@ -83,26 +131,8 @@
-          <kubernetes-security-context-selector component="ctrl.stage.container">
-          </kubernetes-security-context-selector>
-        </stage-config-field>
+        
```
Extraneous space.

danielpeach commented:
One more thing - could you please change the commit message to `feat(provider/kubernetes): ...`? The format is for the changelog generator, which won't recognize `enhance`.

ethanfrogers commented:
@danielpeach sure! sorry about that. will address in a few minutes!

ethanfrogers commented:
@danielpeach i've addressed all of the comments except for wrapping those fields in `col-md-12`. I'm still having trouble with the background color from the collapsible sections bleeding up into those components.

ethanfrogers commented:
i ended up taking a different approach since I couldn't get the `col-md-12` to work. this is closer to the Server Group modal anyway. @danielpeach let me know what you think.

![image](https://user-images.githubusercontent.com/3324110/32845934-060d651c-c9f4-11e7-802e-fa16e287e469.png)
